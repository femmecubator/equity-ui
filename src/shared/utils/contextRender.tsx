import { render, RenderResult } from '@testing-library/react';

import { EquityThemeProvider } from '../../theme';

const contextRender = (ui: React.ReactNode): RenderResult => {
  return render(<EquityThemeProvider>{ui}</EquityThemeProvider>);
};

export default contextRender;
