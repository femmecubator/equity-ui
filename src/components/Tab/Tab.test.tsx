import { describe, expect, it } from 'vitest';
import { screen } from '@testing-library/react';
import contextRender from '../../shared/utils/contextRender';
import TabGroup, { TabItemProps } from './Tab';

describe('Tab component', () => {
  it('TabGroup should render without errors', () => {
    const tabItems: TabItemProps[] = [
      { tabItemId: 1, label: 'one', isActive: true },
      { tabItemId: 2, label: 'two', isActive: false },
      { tabItemId: 3, label: 'three', isActive: false },
    ];

    const handleChange = () => {};

    contextRender(
      <TabGroup tabItems={tabItems} onChange={handleChange}>
        {({ tabItems, onChange }) =>
          tabItems.map((tabItem) => (
            <div
              role="tab"
              key={tabItem.tabItemId}
              onClick={() => onChange(tabItem.tabItemId)}
            >
              {tabItem.label}
            </div>
          ))
        }
      </TabGroup>
    );
    const tabGroupContainer = screen.getByTestId('tab-group-container');
    expect(tabGroupContainer).toBeInTheDocument();

    const tabList = screen.getByRole('tablist');
    expect(tabList).toBeInTheDocument();
    expect(tabList).toHaveAttribute('role', 'tablist');

    const tabLabels = screen.getAllByRole('tab');
    expect(tabLabels).toHaveLength(3);
    expect(tabLabels[0]).toHaveTextContent('one');
    expect(tabLabels[1]).toHaveTextContent('two');
    expect(tabLabels[2]).toHaveTextContent('three');
  });

  it('matches snapshot', () => {
    const tabItems: TabItemProps[] = [
      { tabItemId: 1, label: 'one', isActive: true },
      { tabItemId: 2, label: 'two', isActive: false },
      { tabItemId: 3, label: 'three', isActive: false },
    ];

    const handleChange = () => {};

    const { asFragment } = contextRender(
      <TabGroup tabItems={tabItems} onChange={handleChange}>
        {({ tabItems, onChange }) =>
          tabItems.map((tabItem) => (
            <div
              role="tab"
              key={tabItem.tabItemId}
              onClick={() => onChange(tabItem.tabItemId)}
            >
              {tabItem.label}
            </div>
          ))
        }
      </TabGroup>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
