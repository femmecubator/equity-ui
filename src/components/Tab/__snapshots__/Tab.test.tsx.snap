// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tab component > matches snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  padding-top: 15px;
}

<div
    data-testid="tab-group-container"
  >
    <div
      role="tablist"
    >
      <div
        role="tab"
      >
        one
      </div>
      <div
        role="tab"
      >
        two
      </div>
      <div
        role="tab"
      >
        three
      </div>
    </div>
    <div
      class="emotion-0 emotion-1"
    />
  </div>
</DocumentFragment>
`;
